backend/kernelCI_app/views/treeDetailsBuildsView.py

```diff
@@ -116,3 +117,69 @@ def get(self, request, commit_hash: str | None) -> Response:
             return Response(data=e.json(), status=HTTPStatus.INTERNAL_SERVER_ERROR)
 
         return Response(valid_response.model_dump())
+
+
+class TreeDetailsBuildsDirect(BaseTreeDetailsBuilds):
+    @extend_schema(
+        parameters=[
+            TREE_NAME_PATH_PARAM,
+            GIT_BRANCH_PATH_PARAM,
+            COMMIT_HASH_PATH_PARAM,
+            DirectTreeQueryParameters,
+        ],
+        methods=["GET"],
+        responses=TreeDetailsBuildsResponse,
+    )
+    def get(
+        self,
+        request: HttpRequest,
+        tree_name: str,
+        git_branch: str,
+        commit_hash: str,
+    ) -> Response:
+        try:
+            params = DirectTreeQueryParameters(origin=request.GET.get("origin"))
```

Comment:
This may not have the behavior you expect. Since the `DirectTreeQueryParameters` defines a default option. I think the appropriate way of doing it is using the [model_validate](https://docs.pydantic.dev/latest/concepts/models/#validating-data) method.
```suggestion
            params = DirectTreeQueryParameters.model_validate(request.GET.dict())
```